app/util/util.ts

```diff
@@ -1,10 +1,15 @@
+import { MIST_PER_SUI } from "@mysten/sui/utils";
 import * as bitcoin from "bitcoinjs-lib";
 import Wallet, { Address } from "sats-connect";
 import { fetchUTXOs, fetchValidateAddress } from "~/api/api";
 import { nBTC_ADDR } from "~/constants";
 import { ToastFunction } from "~/hooks/use-toast";
 import { UTXO, ValidateAddressI } from "~/types";
 
+const suiToMist = (amountInSUI: number): bigint => {
+	return BigInt(Math.floor(amountInSUI * Number(MIST_PER_SUI)));
```

Comment:
Ensure you convert amountInSUI to bigint first. Maybe we can get a number overflow error here.

Reply:
Resolving this as discussed.